chain/jsonrpc-primitives/src/types/network_info.rs

```diff
@@ -1,9 +1,15 @@
+use near_network_primitives::types::{KnownProducer, PeerInfo};
 use serde::{Deserialize, Serialize};
 
 #[derive(Serialize, Debug)]
```

Comment:
This now needs to impl `Deserialize`.

Reply:
Could you explain why? We didn't have a Deserialize impl before, so this *shouldn't* break stuff?

Reply:
Currently, `near-jsonrpc-client` uses the `near-client-primitives` version of `RpcNetworkInfoResponse` [(src)](https://github.com/near/near-jsonrpc-client-rs/blob/dd7761b51e1775350be1782370aa22c0b0fe98d7/src/methods.rs#L473). Given that the incoming https://github.com/near/nearcore/pull/5428 patch changes that interface means we have to use this new `near-jsonrpc-primitives` version to maintain compatibility. Meaning, it should be deserializable.

Reply:
@miraclx After we merge #5515, there will be no need to change the `jsonrpc` interface for the renaming.
I don't mind adding Deserialize. However, we don't current use it, and we won't use it after #5428

Reply:
@pmnoxx, Once https://github.com/near/nearcore/pull/5515 is merged, then `NetworkInfoResponse` would no longer be usable from `near-jsonrpc-client` as it would be incompatible for deserializing the server response. We would then have to use this version you just implemented `Deserialize` for. That's why it needs that spec.